src/Interpreters/Context.h

```diff
@@ -261,6 +263,7 @@ class Context: public std::enable_shared_from_this<Context>
 
     /// XXX: move this stuff to shared part instead.
     ContextMutablePtr buffer_context;  /// Buffer context. Could be equal to this.
+    Session * session = nullptr;
```

Comment:
Who owns this session?

Reply:
Somebody else, I had to do this to remove MySQL-specific struct. So the MySQL-stream can access the sequence_id from session (which was previously in Context::mysql)

Reply:
Added a comment to clarify it